CheckMK Installation Error creating credential file

### ✅ Have you read and understood the above guidelines?

yes

### 📜 What is the name of the script you are using?

CheckMK

### 📂 What was the exact command used to execute the script?

bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/ct/checkmk.sh)"

### ⚙️ What settings are you using?

- [ ] Default Settings
- [x] Advanced Settings

### 🖥️ Which Linux distribution are you using?

Debian 12

### 📝 Provide a clear and concise description of the issue.

Install fails with the following error:
```
Processing triggers for php8.2-cgi (8.2.28-1~deb12u1) ...
  ✔️   Installed Checkmk
  ⏳   Creating Service
[ERROR] in line 27: exit code 0: while executing command PASSWORD=$(omd create monitoring | grep "password:" | awk '{print $NF}')

[ERROR] in line 1233: exit code 0: while executing command lxc-attach -n "$CTID" -- bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/install/"$var_install".sh)" $?
```

### 🔄 Steps to reproduce the issue.

install, I ran advanced, but only to configure my network

### ❌ Paste the full error output (if available).

Processing triggers for php8.2-cgi (8.2.28-1~deb12u1) ...
  ✔️   Installed Checkmk
  ⏳   Creating Service
[ERROR] in line 27: exit code 0: while executing command PASSWORD=$(omd create monitoring | grep "password:" | awk '{print $NF}')

[ERROR] in line 1233: exit code 0: while executing command lxc-attach -n "$CTID" -- bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/install/"$var_install".sh)" $?

### 🖼️ Additional context (optional).

_No response_

--- a/install/checkmk-install.sh
+++ b/install/checkmk-install.sh
@@ -28,16 +28,13 @@ SITE_NAME="monitoring4"
 $STD omd create "$SITE_NAME"
 MKPASSWORD=$(openssl rand -base64 18 | tr -d '/+=' | cut -c1-16)
 
-su - "$SITE_NAME" -c "cmk-passwd cmkadmin <<EOF
-$MKPASSWORD
-$MKPASSWORD
-EOF" >/dev/null
+echo -e "$MKPASSWORD\n$MKPASSWORD" | su - "$SITE_NAME" -c "cmk-passwd cmkadmin --stdin"
 $STD omd start "$SITE_NAME"
 
 {
   echo "Application-Credentials"
   echo "Username: cmkadmin"
-  echo "Password: $PASSWORD"
+  echo "Password: $MKPASSWORD"
   echo "Site: $SITE_NAME"
 } >>~/checkmk.creds
 

--- a/install/checkmk-install.sh
+++ b/install/checkmk-install.sh
@@ -24,12 +24,12 @@ motd_ssh
 customize
 
 msg_info "Creating Service"
-PASSWORD=$(omd create monitoring | grep "password:" | awk '{print $NF}')
+MKPASSWORD=$(omd create monitoring | grep "password:" | awk '{print $NF}')
 $STD omd start
 {
     echo "Application-Credentials"
     echo "Username: cmkadmin"
-    echo "Password: $PASSWORD"
+    echo "Password: $MKPASSWORD"
 } >>~/checkmk.creds
 msg_ok "Created Service"
 

--- a/install/checkmk-install.sh
+++ b/install/checkmk-install.sh
@@ -24,7 +24,7 @@ motd_ssh
 customize
 
 msg_info "Creating Service"
-SITE_NAME="monitoring4"
+SITE_NAME="monitoring"
 $STD omd create "$SITE_NAME"
 MKPASSWORD=$(openssl rand -base64 18 | tr -d '/+=' | cut -c1-16)
 

--- a/install/checkmk-install.sh
+++ b/install/checkmk-install.sh
@@ -24,13 +24,23 @@ motd_ssh
 customize
 
 msg_info "Creating Service"
-MKPASSWORD=$(omd create monitoring | grep "password:" | awk '{print $NF}')
-$STD omd start
+SITE_NAME="monitoring4"
+$STD omd create "$SITE_NAME"
+MKPASSWORD=$(openssl rand -base64 18 | tr -d '/+=' | cut -c1-16)
+
+su - "$SITE_NAME" -c "cmk-passwd cmkadmin <<EOF
+$MKPASSWORD
+$MKPASSWORD
+EOF" >/dev/null
+$STD omd start "$SITE_NAME"
+
 {
-    echo "Application-Credentials"
-    echo "Username: cmkadmin"
-    echo "Password: $MKPASSWORD"
+  echo "Application-Credentials"
+  echo "Username: cmkadmin"
+  echo "Password: $PASSWORD"
+  echo "Site: $SITE_NAME"
 } >>~/checkmk.creds
+
 msg_ok "Created Service"
 
 msg_info "Cleaning up"

